plugins/commands/cloud/box/show.rb

```diff
@@ -32,40 +34,47 @@ def execute
                 help: opts.help.chomp
             end
 
-            @client = VagrantPlugins::CloudCommand::Util.client_login(@env, options[:username])
-            box = argv.first.split('/', 2)
+            @client = client_login(@env, options.slice(:quiet))
+            org, box_name = argv.first.split('/', 2)
 
-            show_box(box[0], box[1], options, @client.token)
+            show_box(org, box_name, @client&.token, options.slice(:versions))
           end
 
-          def show_box(org, box_name, options, access_token)
-            username = options[:username]
-
-            server_url = VagrantPlugins::CloudCommand::Util.api_server_url
-            account = VagrantPlugins::CloudCommand::Util.account(username, access_token, server_url)
-            box = VagrantCloud::Box.new(account, box_name, nil, nil, nil, access_token)
-
-            begin
-              success = box.read(org, box_name)
+          # Display the requested box to the user
+          #
+          # @param [String] org Organization name of box
+          # @param [String] box_name Name of box
+          # @param [String] access_token User access token
+          # @param [Hash] options Options for box filtering
+          # @option options [String] :versions Specific verisons of box
+          # @return [Integer]
+          def show_box(org, box_name, access_token, options={})
+            account = VagrantCloud::Account.new(
+              custom_server: api_server_url,
+              access_token: access_token
+            )
+            with_box(account: account, org: org, box: box_name) do |box|
+              if box && options[:versions]
+                box = box.versions.find_all{ |v| options[:versions].include?(v.version) }
+              else
+                box = [box]
+              end
 
-              if options[:version]
-                # show *this* version only
-                results = success["versions"].select{ |v| v if v["version"] == options[:version] }.first
-                if !results
-                  @env.ui.warn(I18n.t("cloud_command.box.show_filter_empty", version: options[:version], org: org, box_name: box_name))
-                  return 0
+              if !box.empty?
+                box.each do |b|
+                  format_box_results(b, @env)
```

Comment:
I think it would be nice to add some whitespace here. Currently if searching for multiple versions of a box the results are stacked right on top of each other eg.

```
Box:          hashicorp/centos-6-dev
Version:      1.0.1603748399
Description:  
Status:       active
Providers:    virtualbox
Created:      2020-10-26T16:49:04.768-05:00
Updated:      2020-10-26T16:49:37.119-05:00
Box:          hashicorp/centos-6-dev
Version:      1.0.1603730619
Description:  
Status:       active
Providers:    virtualbox, vmware_desktop
Created:      2020-10-26T09:59:54.482-07:00
Updated:      2020-10-26T10:00:21.506-07:00
```